### PR TITLE
#213 filter function support.

### PR DESCRIPTION
**Pull request by igalklebanov**
filter functionality Implemented, added to example application and tested.

filter: (time: DateTime, granularity?: 'hours' | 'minutes') => boolean

use case: applications needs to allow selection of only specific hours and/or minutes. some applications (like the one I am working on) have complex scheduling, e.g. 15:30-16:30 and 22:30-23:00, or  2PM/4PM/6PM, and more. current min-max options are not enough in such cases.

### CODE REVIEW

**Comment by Agranom** [comment]
Revert formatting back please

**Comment by igalklebanov** [comment]
> Revert formatting back please

sorry, working on it.

File: src/app/material-timepicker/ngx-material-timepicker.component.ts [hunk 1]
```diff
@@ -73,14 +73,14 @@ export class NgxMaterialTimepickerComponent implements TimepickerRef {
 
     @Input()
     set filter(
-        fn: (time: DateTime, granularity?: "hours" | "minutes") => boolean
+        fn: (time: DateTime, granularity?: 'hours' | 'minutes') => boolean
     ) {
         this._filter = fn;
```
Comment (Agranom) [inline src/app/material-timepicker/ngx-material-timepicker.component.ts]: why do you need setter and getter? Use just `@Input() filter: type`

File: src/app/material-timepicker/services/time-adapter.spec.ts [hunk 2]
```diff
@@ -151,7 +151,7 @@ describe("TimeAdapter", () => {
             expect(isAvailable).toBeTruthy();
         });
 
-        it("should return false since time does not pass filter.", function() {
+        it('should return false since time passes filter.', function() {
```
Comment (Agranom) [inline src/app/material-timepicker/services/time-adapter.spec.ts]: the same test case as above. You might meant `does not pass` ?

Comment (igalklebanov) [reply src/app/material-timepicker/services/time-adapter.spec.ts]: yes :)

File: src/app/material-timepicker/ngx-material-timepicker.component.ts (hunk 1 above)
Comment (igalklebanov) [reply src/app/material-timepicker/ngx-material-timepicker.component.ts]: changed in latest commits.

**Comment by Agranom** [comment]
I'll try to add this feature in the next release

**Comment by Agranom** [comment]
@igalklebanov Pls fix linter's issues

**Comment by Oubayda-Khayata** [comment]
@Agranom Would you please add this feature? Thanks

**Pull request closed without merging**